autogpt/memory/vectara.py

```diff
@@ -0,0 +1,291 @@
+"""Implementation of memory backend using Vectara (https://vectara.com)"""
+import hashlib
+import json
+import logging
+import time
+
+import requests
+from colorama import Fore, Style
+
+from autogpt.config import Config
+from autogpt.logs import logger
+from autogpt.memory.base import MemoryProviderSingleton
+
+
+class JWTTokenProvider:
```

Comment:
can this be abstracted out of just vectara into a more useful global util for jwts?

Reply:
Thanks for the suggestion. I've moved it to `utils.py`.